### PR TITLE
Feat(List-View) : Add the Ability to check if an item is visible or not. 

### PR DESCRIPTION
**Pull request by breningham**
## PR Checklist

- [x] The PR title follows our guidelines: https://github.com/NativeScript/NativeScript/blob/master/CONTRIBUTING.md#commit-messages.
- [ ] There is an issue for the bug/feature this PR is for. To avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it.
- [x] You have signed the [CLA](http://www.nativescript.org/cla).
- [x] All existing tests are passing: https://github.com/NativeScript/NativeScript/blob/master/DevelopmentWorkflow.md#running-unit-tests.
- [x] Tests for the changes are included - https://github.com/NativeScript/NativeScript/blob/master/WritingUnitTests.md.

## What is the current behavior?
There is no way to check if an item is in the ViewPort

## What is the new behavior?
This PR adds a new function for both Android and IOS ListView, that allows checking if the Item is visible or not. 

the function is a member function for the ListView Class, called isItemAtIndexVisible.
and is callable with a reference to the ListView
```(javascript)
ListView.isItemAtIndexVisible(1 /** index of item **/) // returns true/false if item is in view port. 

```

### CODE REVIEW

File: tns-core-modules/ui/list-view/list-view.android.ts [hunk 1]
```diff
@@ -44,9 +47,10 @@ function initializeItemClickListener(): void {
 
 export class ListView extends ListViewBase {
     nativeViewProtected: android.widget.ListView;
-
     private _androidViewId: number = -1;
 
+    public visibleIndexes = [];
```
Comment (vakrilov) [inline tns-core-modules/ui/list-view/list-view.android.ts]: This seems to be unused - can you remove it?

File: tns-core-modules/ui/list-view/list-view.ios.ts [hunk 2]
```diff
@@ -285,6 +285,18 @@ export class ListView extends ListViewBase {
         }
     }
 
+    public isItemAtIndexVisible( itemIndex: number ): boolean {
+        const indexes: NSIndexPath[] = Array.from(this._ios.indexPathsForVisibleRows); 
+
+        for ( const visIndex of indexes ) {
```
Comment (vakrilov) [inline tns-core-modules/ui/list-view/list-view.ios.ts]: Replace the loop with one-liner for readablity:
```
return indexes.some(visIndex => visIndex.row === itemIndex);
```

File: tns-core-modules/ui/list-view/list-view.android.ts [hunk 3]
```diff
@@ -18,6 +19,8 @@ interface ItemClickListener {
     new (owner: ListView): android.widget.AdapterView.OnItemClickListener;
 }
 
+
```
Comment (vakrilov) [inline tns-core-modules/ui/list-view/list-view.android.ts]: Causes a [tslint error: Consecutive blank lines are forbidden](https://travis-ci.org/NativeScript/NativeScript/builds/354457176#L1010)

File: tests/app/ui/list-view/list-view-tests.ts [hunk 4]
```diff
@@ -705,6 +705,32 @@ export class ListViewTest extends testModule.UITest<listViewModule.ListView> {
...
+    public test_check_if_item_at_index_is_visible() {
+        var listView = this.testView;
+
+        listView.itemTemplate = () => {
+            var label = new Label();
+            label.id = "testLabel";
+            label.bind({ sourceProperty: "$value", targetProperty: "text", twoWay: false });
+            return label;
+        }
+        listView.items = [1, 2, 3];
```
Comment (vakrilov) [inline tests/app/ui/list-view/list-view-tests.ts]: Can you make the items array bigger (probably 30-40 items will be enough) and assure the item on the last index is *not* visible too.

**Comment by vakrilov** [comment]
Hey @bergenhem, thanks for the awesome PR! I have requested some minor code changes. As soon as they are ready and all is green - this is ready for merge.

**Comment by breningham** [comment]
I think you've tagged the wrong guy! Haha
I'll get on them changes today.

**Comment by breningham** [comment]
I have updated the pull request with the requested changes.

**Comment by vakrilov** [comment]
test

**Comment by vakrilov** [comment]
Merged! Thanks for the awesome PR!

**Pull request merged**